[FLINK-2752] [Documentation] Documentation is not easily differentiable from the Flink homepage

mjsax opened the pull request:
Changed menu color schema to make page clearly distinguishable from project page.

mxm commented on docs/internals/coding_guidelines.md:
```diff
@@ -1,23 +0,0 @@
-under the License.
--->
-
-The *Coding Guidelines* are located [on the project website](http://flink.apache.org/coding-guidelines.html).
```
I think it's fine to keep these for a while to avoid dead links.

mjsax replied:
Fair enough. What about auto-redirect? (same below of course)

mxm replied:
Yes. Using meta refresh? Why not.

mxm commented on docs/internals/how_to_contribute.md:
```diff
@@ -1,23 +0,0 @@
-under the License.
--->
-
-The *How to Contribute* guide is located [on the project website](http://flink.apache.org/how-to-contribute.html).
```
I think it's fine to keep these for a while to avoid dead links.

mxm commented on docs/_includes/navbar.html:
```diff
@@ -39,7 +39,7 @@
         <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
           <ul class="nav navbar-nav">
-            <li{% if page.url == '/index.html' %} class="active"{% endif %}><a href="{{ site.baseurl}}/index.html">Overview<span class="hidden-sm hidden-xs"> {{ site.version_short }}</span></a></li>
+            <li{% if page.url == '/' %} class="active"{% endif %}><a href="{{ site.baseurl}}/">Overview<span class="hidden-sm hidden-xs"> {{ site.version_short }}</span></a></li>
```
Can we change "Overview" to "Documention"?

mjsax replied:
We can for sure. But do we want? :) Why should it be better? I think "Overview" is good.

mxm commented:
I'm still in favor of a "Back to Flink homepage" link. The navigation bar color looks quite dull to me.

<img width="1440" alt="screen shot 2015-10-11 at 16 15 31" src="https://cloud.githubusercontent.com/assets/837221/10909276/2c3e3424-8238-11e5-90d3-ac65caa3d531.png">

mjsax commented:
Wrong screenshot? I am open to use different colors for the menu -- I am not a designer and this was just a try to come up with something...

I think a "Back to Flink homepage" link would be nice, too. But do we really want to change the width of the whole page? (As far as I understand, the width of the menu should be the same width as the page width.) I adden "Flink Homepage" menu point, making the menu too wide. Thus, the search input filed and search button got rendered in a second line, making the menu bar twice as high and breaking the layout...

mxm commented:
Maybe @uce can help us here. He designed the website and documentation layout.

What about adding a bar on top of the navigation bar which contains a link back to the homepage and the version of the documentation?

mxm commented:
I updated the screenshot.

mjsax commented:
Just updated this. Redirect works now for contribution/coding guidelines. About the color scheme: it is not very exciting; however, the purpose was to make it better distinguishable from the Flink project page -- I guess this goal is reached. If nobody has a better ideas, I would like to merge this.

aljoscha commented:
We should not just merge it without addressing peoples concerns about the color scheme. IMHO it does not look good like this and there are also issues with the highlighting of the currently selected page. For example, if I click "Overview 1.0" the menu entry is not highlighted, as it is before this change.

Also the mouse-over color is now different from the colors that are used to signify the currently selected page.

mjsax commented:
Of I want to address other peoples concerns: "If nobody has a better ideas".
What would be a good color scheme? Please provide some constructive ideas! I will not merge as long as people tell me, they don't like it. But "I don't like it" does not help to improve it ;)

aljoscha commented:
I would simply change "Overview 1.0" to "Documentation 1.0" in the doc.
Doc:
![screen shot 2015-11-04 at 17 22 06](https://cloud.githubusercontent.com/assets/68551/10944283/cb88d3d2-8318-11e5-8251-b8e3513be84c.png)
Flink Website:
![screen shot 2015-11-04 at 17 22 12](https://cloud.githubusercontent.com/assets/68551/10944351/1278e3f4-8319-11e5-9da0-2cdbfddc886f.png)

To me this makes it clear that I'm in the documentation. I can click the Flink Logo to go back to the Flink homepage.

mxm commented:
I second Aljoscha. Also, I suggested twice (on the mailing list and in the PR comments) to change "Overview" to "Documentation" which should help a lot to differentiate the pages.

mxm commented:
The grey navigation bar is not an acceptable solution to me. Furthermore, I suggested to add a bar on top of the navigation with something like "Documentation 1.0 | Back to Flink homepage".

mjsax commented:
I don't like the idea of the second navigation bar. The menu should not occupy too much space. And using the logo as back link to the homepage is a very good solution. @aljoscha What is your opinion?

About the color: I player with a view other colors, and all of them were ugly... What color scheme would you like? It was gray before -- so is your concern about gray in general or about the darker color?

mxm commented:
Why do you open a pull request against a JIRA issue if you don't plan to address its concerns and ideas? If you don't agree with the JIRA description, please feel free to start a discussion in JIRA before opening a pull request.

-1 from my side for a different color scheme. The colors are well-adjusted and don't need to be changed.

mjsax commented:
I guess there is some miss understanding. Of course I want to address concerns and ideas. That is the reason why we have this discussion (and I am glad that you participate!). The JIRA itself is not very precise and only states an idea how it could be resolved. This PR is my suggestion how to resolve it. If the community does not like it, that's fine. Up to now, I interpreted your concerns about the changing the color as "you don't like *how* I changed the colors" -- now it got clear to me, that you don't like to idea of changing color at all. (I miss understood your concerns; sorry for that).

Furthermore, I did start a discussion about this on the mailing list and @uce had the idea about changing the style: https://mail-archives.apache.org/mod_mbox/flink-dev/201510.mbox/%3C79C15FB9-EB82-433F-AC1D-2823CEF018AE%40apache.org%3E
Some people seems to like it and I shared a like to a first draft (before I opened this PR). There was not too much feedback about the link, so I proceeded by opening PR.

I take your concerns very seriously but still like the idea about changing the style of the doc page for a clearer distinction from the project web page.

I guess we all agree that changing "Overview" to "Documentation" should be done. But This does not resolve the JIRA IMHO.

mxm commented:
Thanks for clarifying your approach on the issue. I believe that you are taking concerns seriously but I also think that merging when there are clear open concerns is not an option. 

I don't want to dismiss the option of changing the color scheme per se but I don't want to sacrifice readability of the documentation just to make it appear differently from the homepage. So my take on the issue would be to change only very small things like replacing "Overview" with "Documentation" and including a hint for the user that he can go back to the homepage. For design changes, I'd like someone to work on it who knows how to change the style in a nice way.

What about a notice text on the front page of the documentation like "You're viewing the documentation 1.0. To go back to the Flink homepage please click on the squirrel." That  might already help a lot of people.

mjsax commented:
I agree that changing the colors is a tricky issues. Does anybody have good design skills? From my point of view, a clear visual difference is the only way to get a clear distinction. All other suggestion are only minor improvements of the current status.

I tried to "play designer" but without success (even if I like it personally and don't thinks I sacrificed readability -- but of course, this is just personal opinion). As I am running out of further ideas, I guess somebody else needs to resolve the JIRA.

If you are ok with it, I will change "Overview" to "Documentation" as well as the redirect thing and merge those changes to close this PR. The JIRA stays open...

One more thing: I also had the idea of adding the "external link" symbol at the main page to the "Quickstart" and "Documentation" menu (either in the menu point or in all link of the menu-list). I did not got any feedback if this would be an improvement or not (I am not sure by myself either). Of course, this does not reflect this PR but is a change in flink-web project.

mxm commented:
I'm with you that the design could be improved but I rather have some small changes such that we have an improvement over the current situation. So let's merge with "Overview" renamed to "Documentation", redirect from the How-to-contribute page, and adding external link icons to the Documentation and Quickstart menu (good idea!).

mjsax commented:
Changing "Overview" to "Documenation" breaks the layout for me (using Firefox). What about using "Home Docu 1.0" instead?

![screenshot](https://cloud.githubusercontent.com/assets/8959638/10974851/36a3c804-83e4-11e5-9a74-497c42f77a66.png)

mxm commented:
I'm not too fond of "Home Docu". Can you reduce the size of the search box? That should do it.

mjsax commented:
Good idea. Just updated this. Please verify that it renders correctly for you. And let me know if I can merge it.

mxm commented on docs/internals/coding_guidelines.md:
```diff
@@ -20,4 +20,6 @@ specific language governing permissions and limitations
 under the License.
 -->
 
+<meta http-equiv="refresh" content="0; url=http://flink.apache.org/contribute-code.html#coding-guidelines" />
```
Maybe increase the timeout a bit. Otherwise it might be a bit confusing.

mxm commented on docs/internals/how_to_contribute.md:
```diff
@@ -20,4 +20,6 @@ specific language governing permissions and limitations
 
-The *How to Contribute* guide is located [on the project website](http://flink.apache.org/how-to-contribute.html).
\ No newline at end of file
+<meta http-equiv="refresh" content="0; url=http://flink.apache.org/how-to-contribute.html" />
```
Same here.

mxm commented:
Thanks for the update. Works for me in Firefox, Chromium, and Safari. Good to merge with an increased timeout for the redirect.

mjsax commented:
Just updated. Set redirect to 5 seconds (this gives enough time to read the single line). Also added comment that auto redirect is enabled. Please check the "look an feel" of the redirect.

mxm commented:
Looks good to merge to me.

mjsax commented:
Merging this, together with https://github.com/apache/flink-web/pull/13 and https://github.com/apache/flink-web/pull/12
